File: iOS/MarvelApp/Services/Utils/SortingOrder.swift
//
//  SortingOrder.swift
//  MarvelApp
//
//  Created by Suyash Srijan on 20/04/2018.
//  Copyright © 2018 Suyash Srijan. All rights reserved.
//

import Foundation

enum SortingOrder: String {
    case FOC_DATE = "focDate"
    case ON_SALE_DATE = "onsaleDate"
    case TITLE = "title"
    case ISSUE_NUMBER = "issueNumber"
    case MODIFIED = "modified"
    
    case REVERSED_FOC_DATE = "-focDate"
    case REVERSED_ON_SALE_DATE = "-onsaleDate"
    case REVERSED_TITLE = "-title"
    case REVERSED_ISSUE_NUMBER = "-issueNumber"
    case REVERSED_MODIFIED = "-modified"
}

File: Android/app/src/main/kotlin/com/suyashsrijan/marvel/app/network/enums/SortingOrder.kt
package com.suyashsrijan.marvel.app.network.enums

enum class SortingOrder(val value: String) {

    FOC_DATE("focDate"),
    ON_SALE_DATE("onsaleDate"),
    TITLE("title"),
    ISSUE_NUMBER("issueNumber"),
    MODIFIED("modified"),
    REVERSE_FOC_DATE("-focDate"),
    REVERSE_ON_SALE_DATE("-onsaleDate"),
    REVERSE_TITLE("-title"),
    REVERSE_ISSUE_NUMBER("-issueNumber"),
    REVERSE_MODIFIED("-modified");

}